js/components/graph/Graph.js

```diff
@@ -494,15 +501,15 @@ export class Graph extends React.Component {
       if (this.state.draggingNode !== null) {
         const newPos = this.getRelativeCoords(event)
         let currentNode
-        for (const node of this.state.nodesJSON) {
+        for (const node of Object.values(this.state.nodesJSON)) {
           if (node.id_ === this.state.draggingNode) {
             currentNode = node
           }
         }
         currentNode.pos = [newPos.x - 20, newPos.y - 15]
         currentNode.text[0].pos = [newPos.x, newPos.y + 5]
-        const newNodesJSON = [...this.state.nodesJSON]
-        newNodesJSON.push(currentNode)
+        const newNodesJSON = [...Object.values(this.state.nodesJSON)]
```

Comment:
You should be creating an object, not an array here. You can try using just `{...this.state.nodesJSON}` instead. This occurs a few more times below.